ai-safety-debate/train_classifier_via_debate.py

```diff
@@ -91,26 +120,35 @@ def run(
             sample = train_data[i]
             probs = next(debate_classifier.predict(sample))["probabilities"]
             label = np.random.choice(range(len(probs)), p=probs)
-            # label = judge.train_labels[i]
-            probs[label] = 0
-            probs /= probs.sum()
-            # print("i", i, "label", label)
-            # print("i", i, "label2", label2)
+            restricted_first = np.random.random() < 0.5
 
             if cheat_debate:
                 # simulate a perfectly accurate debate
                 utility = -1 if label == judge.train_labels[i] else 0
-            elif debate_results is not None:
-                # use precompuded results
-                probabilities = debate_results[i, label]
-                if np.all(probabilities[label] >= probabilities):
+            elif debate_results_restricted_first is not None:
+                assert debate_results_restricted_second is not None
+                if restricted_first:
+                    debate_results = debate_results_restricted_first
+                else:
+                    debate_results = debate_results_restricted_second
+                # use precomputed results
+                judge_probabilities = debate_results[i, label]
+                if np.all(judge_probabilities[label] >= judge_probabilities):
                     utility = -1
                 else:
                     utility = 0
             else:
                 # run non-precommited debate
-                agent1 = DebateAgent(precommit_label=None, agentStrength=rollouts)
-                agent2 = DebateAgent(precommit_label=label, agentStrength=rollouts)
+                agent_unrestricted = DebateAgent(
+                    precommit_label=None, agentStrength=rollouts
+                )
+                agent_restricted = DebateAgent(
+                    precommit_label=label, agentStrength=rollouts
+                )
+                if restricted_first:
+                    agent1, agent2 = agent_restricted, agent_unrestricted
+                else:
+                    agent1, agent2 = agent_unrestricted, agent_restricted
                 debate = Debate((agent1, agent2), judge, N_to_mask, sample.flat)
                 utility = debate.play()
```

Comment:
I don't really understand how the mapping between wins and utility goes, but I'd guess that the reason that utility=-1 -> win and utility=0 -> loss is that the honest agent always played last, previously? If so, that's not going to hold up any more, because sometimes the honest player plays first (in the line above), in which case utility being 0 or 1 corresponds to it winning?